### PR TITLE
Add migration guide for 2.0

### PR DESCRIPTION
**Pull request by MusicDin**
Adds migration guide for release 2.0.

### CODE REVIEW

File: docs/guides/migration_2.0.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- `lxd_snapshot.creation_date` - Replaced by `lxd_snapshot.created_at`.
+
+## Renamed attributes and resources
+
+The major release also provided an opportunity to rename certain attributes for consistency across
+the provider.
+
+List of renamed attributes:
+- `lxd_file.source` -> `lxd_file.source_path`
+- `lxd_file.target_file` -> `lxd_file.target_path`
```
Comment (simondeziel) [inline docs/guides/migration_2.0.md:45]: I could only find `lxd_instance_file`, not `lxd_file`.

**Review by simondeziel — COMMENTED** [review]
Just a quick question that needs to be clarified before merging, thanks!

File: docs/guides/migration_2.0.md (hunk 1 above)
Comment (MusicDin) [reply docs/guides/migration_2.0.md:45]: Good spot. Thanks :)

File: docs/guides/migration_2.0.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## Renamed attributes and resources
+
+The major release also provided an opportunity to rename certain attributes for consistency across
+the provider.
+
+List of renamed attributes:
+- `lxd_instance.file.instance_name` -> `lxd_instance.file.instance`
+- `lxd_instance.file.source` -> `lxd_instance.file.source_path`
+- `lxd_instance.file.target_file` -> `lxd_instance.file.target_path`
```
Comment (simondeziel) [inline docs/guides/migration_2.0.md:46]: I think those are typo'ed (`lxd_instances.file.*`) version of `lxd_instance_file.*` ones further down.

File: docs/guides/migration_2.0.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- `lxd_instance.start_on_create` -> `lxd_instance.running`
```
Comment (simondeziel) [inline docs/guides/migration_2.0.md:47]:
```suggestion
- `lxd_instance.start_on_create` -> `lxd_instance.running`
- `lxd_instance_file.instance_name` -> `lxd_instance_file.instance`
```

File: docs/guides/migration_2.0.md (hunk 2 above)
Comment (MusicDin) [reply docs/guides/migration_2.0.md:46]: Both are modified - `lxd_instance.file` (file block of lxd_instance resource) and `lxd_instance_file` resource

**Review by simondeziel — APPROVED** [review]

File: docs/guides/migration_2.0.md (hunk 2 above)
Comment (simondeziel) [reply docs/guides/migration_2.0.md:46]: Sounds good, thanks!

Comment (MusicDin) [reply docs/guides/migration_2.0.md:46]: But, `lxd_instance.file.instance_name` is indeed wrong hehe. Should be `lxd_instance_file.instance_name`

**Pull request merged**